sunpy/image/tests/test_transform.py

```diff
@@ -319,32 +319,38 @@ def test_nans(rot30):
     image_with_nans[4:-4, 4:-4] = 2
     image_with_nans[9:-9, 9:-9] = np.nan
 
-    fig = Figure(figsize=(16, 4))
-    axs = fig.subplots(2, 7)
+    num_methods = len(_rotation_registry.keys())
 
-    axs[0, 0].set_title('Original (NaNs are white)')
-    axs[0, 0].imshow(image_with_nans, vmin=-1.1, vmax=1.1)
-    axs[1, 0].imshow(image_with_nans, vmin=-1.1, vmax=1.1)
+    fig = Figure(figsize=(16, 2*num_methods))
+    axs = fig.subplots(num_methods, 7)
 
-    for i in range(6):
-        axs[0, i+1].set_title(f'order={i}')
-        axs[0, i+1].imshow(affine_transform(image_with_nans, rot30,
-                                            order=i, method='scipy', missing=np.nan),
-                           vmin=-1.1, vmax=1.1)
-        axs[1, i+1].imshow(affine_transform(image_with_nans, rot30,
-                                            order=i, method='skimage', missing=np.nan),
-                           vmin=-1.1, vmax=1.1)
+    axs[0, 0].set_title('Original (NaNs are white)')
 
-    axs[0, 0].set_ylabel('SciPy')
-    axs[1, 0].set_ylabel('scikit-image')
+    for j in range(6):
+        axs[0, j+1].set_title(f'order={j}')
+    for i, method in enumerate(_rotation_registry.keys()):
+        axs[i, 0].imshow(image_with_nans, vmin=-1.1, vmax=1.1)
+        for j in range(6):
+            if j not in _rotation_registry[method].allowed_orders:
+                with pytest.raises(ValueError):
+                    affine_transform(image_with_nans, rot30, order=j, method=method, missing=np.nan)
+                axs[i, j+1].remove()
+            else:
+                axs[i, j+1].imshow(affine_transform(image_with_nans, rot30,
+                                                    order=j, method=method, missing=np.nan),
+                                   vmin=-1.1, vmax=1.1)
+        axs[i, 0].set_ylabel(method)
 
     return fig
 
 
 @pytest.mark.filterwarnings("ignore:.*bug in the implementation of scikit-image")
-@pytest.mark.parametrize('method', _rotation_function_registry.keys())
+@pytest.mark.parametrize('method', _rotation_registry.keys())
 @pytest.mark.parametrize('order', range(6))
 def test_endian(method, order, rot30):
+    if order not in _rotation_registry[method].allowed_orders:
+        return
```

Comment:
Do a `pytest.skip` with a suitable message here instead?

Reply:
Hmm, yes, that could be done, but wouldn't that just add noise to the short summary of every test run?  There'd always be skips reported because these orders are simply not supported.  I'm inclined to consider them passes rather than skips.

Reply:
I am with Albert on this, adding more skip test results at the end of the test output is not something I support especially since we might never be able to unskip them.